### PR TITLE
Refactor Version to VersionModel

### PR DESCRIPTION
**Pull request by dkatzz**
### Description

Refactor the "Version" class to be "VersionModel", so it is clear when we may be accessing the database.

We'll do this to the other Models as well as a follow up.

Similar to https://github.com/civiform/civiform/pull/5960

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers).

### Issue(s) this completes
 
#5961

### CODE REVIEW

File: server/test/models/VersionModelTest.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
 import services.question.exceptions.QuestionNotFoundException;
 
 // TODO(#4912): Add more tests for Version.java.
```
Comment (caitlinshk) [inline server/test/models/VersionModelTest.java:16]: Nit: Update this comment too?

**Review by caitlinshk — APPROVED** [review]

File: server/test/models/VersionModelTest.java (hunk 1 above)
Comment (dkatzz) [reply server/test/models/VersionModelTest.java]: Good call - updated!

File: server/test/controllers/monitoring/MetricsControllerTest.java [hunk 2]
```diff
@@ -62,7 +62,7 @@ public void getMetrics_returnsMetricData() {
     assertThat(metricsContent).contains("ebean_queries_total_latency_micros");
     assertThat(metricsContent).contains(getEbeanCountName("models.ProgramModel"));
     assertThat(metricsContent).contains(getEbeanCountName("models.Question"));
-    assertThat(metricsContent).contains(getEbeanCountName("Version.byId"));
+    assertThat(metricsContent).contains(getEbeanCountName("VersionModel.byId"));
     assertThat(metricsContent).contains("location=\"VersionRepository.getActiveVersion");
```
Comment (zarifzm) [inline server/test/controllers/monitoring/MetricsControllerTest.java:66]: This statement causes the test to fail. Seems to indicate there's something else to refactor in the codebase?

Comment (dkatzz) [reply server/test/controllers/monitoring/MetricsControllerTest.java:66]: Thank you! Yeah - just fixed the MetricsController and this now passes.

**Pull request merged**